extensions/amp-subscriptions/0.1/renderer.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {dev} from '../../../src/log';
+
+const CSS_PREFIX = 'i-amphtml-subs';
+
+
+export class Renderer {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    /** @const @private */
+    this.ampdoc_ = ampdoc;
+
+    /** @const @private {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(ampdoc.win);
+
+    // Initial state is "unknown".
+    this.setGrantState_(null);
+    this.setLoggedinState_(null);
+    this.setSubscriberState_(null);
+    this.getRootElement_().classList.add(`${CSS_PREFIX}-ready`);
+  }
+
+  /**
+   * @return {!Element}
+   * @private
+   */
+  getRootElement_() {
+    const root = this.ampdoc_.getRootNode();
+    return dev().assertElement(root.documentElement || root.body || root);
+  }
+
+  /**
+   * @param {string} type
+   * @param {?boolean} state
+   * @private
+   */
+  setState_(type, state) {
+    this.vsync_.mutate(() => {
+      this.getRootElement_().classList.toggle(
+          `${CSS_PREFIX}-${type}-unk`,
```

Comment:
is `unk` something we regularly use across AMP?
can we make it `unknown/none`?

Reply:
This is a private class - not user visible. So, I'm ok keeping it slightly obscure. But if you feel strongly about it we expand.